File: app/src/main/java/com/billiamram/hexagonalscaffold/AndroidModule.java
package com.billiamram.hexagonalscaffold;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidModule {

    private Context context;

    public AndroidModule(Context context) {
        this.context = context;
    }

    @Singleton
    @Provides
    SharedPreferences provideSharedPreferences() {
        return context.getSharedPreferences("default", 0);
    }

}

File: app/src/main/java/com/billiamram/hexagonalscaffold/AndroidModule.kt
package com.billiamram.hexagonalscaffold

import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AndroidModule(private val context: Context) {

    @Singleton
    @Provides
    internal fun provideSharedPreferences(): SharedPreferences {
        return context.getSharedPreferences("default", 0)
    }

}
